ioctl/cmd/bc/bcdelegate.go

```diff
@@ -142,3 +150,77 @@ func getDelegateByName(name string) (*iotextypes.CandidateV2, error) {
 	}
 	return &delegate, nil
 }
+
+func getDelegateByAddress(addr string) (*iotextypes.CandidateV2, error) {
+	conn, err := util.ConnectToEndpoint(config.ReadConfig.SecureConnect && !config.Insecure)
+	if err != nil {
+		return nil, output.NewError(output.NetworkError, "failed to connect to endpoint", err)
+	}
+	defer conn.Close()
+	cli := iotexapi.NewAPIServiceClient(conn)
+
+	readCandidatesLimit := 20000
+	for i := uint32(0); ; i++ {
+		offset := i * uint32(readCandidatesLimit)
+		size := uint32(readCandidatesLimit)
+		candidateList, err := getDelegates(cli, offset, size)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get candidates")
+		}
+		if idx := slices.IndexFunc(candidateList.Candidates, func(cand *iotextypes.CandidateV2) bool {
+			return cand.OperatorAddress == addr
+		}); idx >= 0 {
+			return candidateList.Candidates[idx], nil
+		}
+		if len(candidateList.Candidates) < readCandidatesLimit {
+			break
+		}
+	}
+	return nil, output.NewError(output.UndefinedError, "failed to find delegate", nil)
+}
+
+func getDelegates(cli iotexapi.APIServiceClient, offset, limit uint32) (candidateList *iotextypes.CandidateListV2, err error) {
```

Comment:
there is already `func getStakingCandidates()` in `nodedelegate.go`
move it to /util so can re-use the code?

Reply:
good advice